### DISCUSSION TITLE
How to read cookies in the request header?

### DISCUSSION THREAD

**Discussion by lgharibashvili** [MichalLytek/type-graphql#1307] [Q&A]
Is it possible to access request object from resolver or middleware function?

I would like to read cookies from the request header for cookie-based authentication.

**Comment by EmirBoyaci** [comment]
Do you really want to access request headers in resolver or while context creation so that you make sure only authenticated users can execute operations in graphql. Then, you can put `@Authorized()` decorators on top of your query, mutation or subscriptions so that you can make sure authenticated user has specified role(s) with custom `authChecker`. For doing what I said: see my reply in other discussion: https://github.com/MichalLytek/type-graphql/discussions/1308#discussioncomment-3973425
